src/app/modules/core/core.component.ts

```diff
@@ -9,14 +9,33 @@ import { Router } from '@angular/router';
 })
 export class CoreComponent implements OnInit {
     title = 'Inbox';
+    login_time: any;
+    current_time: any;
+    diff: any;
     constructor(private _router: Router) { }
 
     ngOnInit(): void {
         this.title = 'Inbox';
+        if (localStorage.getItem('token') !== '') {
```

Comment:
this is incorrect logic. 

you don't need to calculate time, 
instead file an api /verify?access_token=

which will verify if access token is valid or not